src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -5,36 +5,62 @@
-        self.parser.add_argument('destination_folder', help='root folder of created project', type=pathlib.Path)
+        self.parser.add_argument('project_name', help='project name')
+        self.parser.add_argument('input_file', help='input file')
+        self.parser.add_argument('destination_folder', help='root folder of created project', type=pathlib.Path,
+                                 nargs='?')
         self.parser.add_argument('-l', '--layout', action='store', dest='layout',
                                  choices=['default', 'simulator'], default='default',
                                  help='Project layout: default (in place) or simulator (one folder per library with linked files)')
+        # Note on --uvm :
+        #  if --uvm is not on the command line, None is returned
+        #  if --uvm is on the command line without a path, 'use_UVM_HOME' is returned
+        #  if --uvm=path is given, the path is returned
+        self.parser.add_argument('--uvm', help='add UVM to the project. optionally add the path to your UVM folder as an argument',
+                                 dest='uvm', nargs='?', const='use_UVM_HOME')
+        self.parser.add_argument('--uvmlib', help='library in which to compile the UVM package (default `work`)',
+                                 dest='uvmlib', default='work')
 
     def parse_args(self):
         args = self.parser.parse_args()
         if not os.path.isfile(args.input_file):
-            self.parser.error("Incorrect number of arguments: given input file does not exist")
+            self.parser.error('Input file does not exist')
         if args.destination_folder is not None:
             if not args.destination_folder.is_dir():
-                self.parser.error("destination folder has to be a folder")
+                self.parser.error('destination folder has to be a folder')
+        if args.uvm == 'use_UVM_HOME':
+            args.uvm = os.environ.get('UVM_HOME')
+            if args.uvm is None:
+                self.parser.error('UVM_HOME must be set when using --uvm without an argument')
+        if args.uvm is not None:
+            if not os.path.isdir(args.uvm):
+                self.parser.error(f'UVM home \'{args.uvm}\' (from command line or UVM_HOME) must point to a folder')
+            if not os.path.isfile(os.path.join(args.uvm, 'src/uvm_macros.svh')):
+                self.parser.error(f'Could not find uvm_macros.svh in \'{args.uvm}/src\'')
+            if not os.path.isfile(os.path.join(args.uvm, 'src/uvm_pkg.sv')):
+                self.parser.error(f'Could not find uvm_pkg.sv in \'{args.uvm}/src\'')
         return args
 
     def parse_args_and_file(self, parse_file):
         args = self.parse_args()
         ArgsAndFileParser.options = args
         destination = args.destination_folder if args.destination_folder is not None else os.getcwd()
         entries = parse_file(args.input_file)
+        print(f'*debug* UVM options: {args.uvm} {args.uvmlib}')
```

Comment:
Debug prints should use `print(..., file=stderr)` (with `from sys import stderr`) to avoid confusion with program output. (Assuming the rest of the script does this correctly)

Reply:
Debug code removed